components/basecomponents/StoryReader.tsx

```diff
@@ -24,9 +24,19 @@ const StoryReader = ({ titleCurrent, titleOther, id, phrases }: StoryReaderProps
     useStoryReader(id);
 
   const handleLanguageChange = (language: Language) => {
-    setSeekValue(0);
-    setLanguagePlay(language);
-    stopStory();
+    const ukCurrent = language === 'uk';
+    const currentTime = audio.current?.currentTime || 0;
+    if ((ukCurrent && currentTime < phrases[0].start_uk) || (!ukCurrent && currentTime < phrases[0].start_cs)) {
```

Comment:
I don't understand the purpose of this condition, why special treatment of the first phrase? Why not apply the same logic as for all other phrases? :) 

Otherwise it looks good and works :)

Reply:
The first phrase has to be treated differently, because it starts at about 15 seconds, so if you would click the button for changing language in let's say first 10 seconds, it would send you to the fifteenth second, which is not what we want.

Reply:
Should I add some comments right into the code so that it is more understandable?